charts/trustify/templates/helpers/_infrastructure.tpl

```diff
@@ -44,6 +44,8 @@ Arguments (dict):
   value: parentbased_traceidratio
 - name: OTEL_TRACES_SAMPLER_ARG
   value: "0.1"
+- name: OTEL_EXPORTER_OTLP_ENDPOINT
+  value: http://jaeger{{ .root.Values.appDomain }}
```

Comment:
Ah great!

Can we introduce a new `define` for this. Maybe in the endpoints. And allow overriding this default from the `tracing` field/object in the values file?

Reply:
:shrug: 

I have no idea what what you are talking about :grin: 

enabling this ... apparently ... trustify-minikube -- can send to --> jaeger-minikube as I understood 'localhost' doesn't exist inside that environment [??] . but we end up with `received message with invalid compression flag: 60 (valid flags are 0 and 1)` which apparently is a thing caused/handled by `tonic` https://github.com/helio-frota/otel-actix-example/blob/main/src/otel.rs#L27 , and then:

* Why this is happening inside minikube?
* Why we can't see this compression error when sending traces from an app running outside minikube?

Reply:
Ok, my proposal is, instead of hardcoding `http://jaeger{{ .root.Values.appDomain }}`, we'd need some way to provide a link to the jaeger `Service` (cluster internal) address, not necessarily the `Ingress` (cluster external) address. Technically ingress works too, but that might mean all traffic goes through the external load balancer. Which is unnecessary.

If the jaeger instance is in the same namespace, something like `http://jeager` (where `jaeger` is the name of the `Service` object) would work. If it is in a different k8s namespace, then we need something like `http://<jaeger.><namespace>.svc.cluster.local` (where `jaeger` is the `Service` name, and `namespace` the namespace).

However, if the instance is something completely different, we should support that as well. Meaning, allow the user to provide the full endpoint and just accept is as it. If we can provide a static default in the values files, helm already provides us with the override. So we should:

* [ ] Define a reasonable default in the values file, based on the setup of the infrastructure helm chart. e.g. in `.tracing.jaegerEndpoint`.
* [x] Allow the user to override that value through helm

In order to not hardcode this in every place, we should:

* [ ] `define` a template for this in e.g. `_tracing.tpl`: https://helm.sh/docs/chart_template_guide/named_templates/
* [ ] use that template instead of `http://jaeger{{ .root.Values.appDomain }}`

Reply:
you right, the pods are in the same namespace `trustify` and we can't hardcode this `http://jaeger`
Thanks for the clarification :+1: